src/WallGo/config.py

```diff
@@ -0,0 +1,244 @@
+    """ Maximum temperature that is probed in Hydrodynamics (in units of Tnucl). """
+    
+    relativeTol: float = 1e-6
+    """ Relative tolerance used in Hydrodynamics. """
+    
+    absoluteTol: float = 1e-10
+    """ Absolute tolerance used in Hydrodynamics. """
+
+@dataclass
+class ConfigThermodynamics:
+    """ Holds the config of the Hydrodynamics class. """
+    
+    tmin: float = 0.8
+    """
+    Minimum temperature used in the phase tracing (in units of the estimate for the
+    minimum temperature obtained in the template model). 
+    """
+    
+    tmax: float = 1.2
+    """
+    Maximum temperature used in the phase tracing (in units of the estimate for the
+    maximum temperature obtained in the template model). 
+    """
+
+    phaseTracerTol: float = 1e-6
+    """
+    Desired accuracy of the phase tracer and the resulting FreeEnergy interpolation.
+    """
+
+@dataclass
+class ConfigBoltzmannSolver:
+    """ Holds the config of the BoltzmannSolver class. """
+    
+    basisM: str = 'Cardinal'
+    """ The position polynomial basis type, either 'Cardinal' or 'Chebyshev'. """
+    
+    basisN: str = 'Chebyshev'
+    """ The momentum polynomial basis type, either 'Cardinal' or 'Chebyshev'. """
+    
+    collisionMultiplier: float = 1.0
+    """
+    Factor multiplying the collision term in the Boltzmann equation. Can be used for
+    testing or for studying the solution's sensibility to the collision integrals. Don't
+    forget to adjust meanFreePathScale accordingly if this is different from 1
+    (meanFreePathScale should scale like 1/collisionMultiplier).
+    WARNING: THIS CHANGES THE COLLISION TERMS WRT TO THEIR PHYSICAL VALUE.
+    """
+
+@dataclass
+class Config:
+    """
+    Data class that holds all the model-independent configs.
+    It contains objects of the data classes ConfigGrid, ConfigEOM,
+    ConfigEffectivePotential, ConfigHydrodynamics, ConfigThermodynamics and
+    ConfigBoltzmannSolver.
+    It can also load the configs from an .ini file.
+    """
+
+    configGrid: ConfigGrid = ConfigGrid()
+    """ Holds the config of the Grid3Scales class. """
+
+    configEOM: ConfigEOM = ConfigEOM()
+    """ Holds the config of the EOM class. """
+
+    configHydrodynamics: ConfigHydrodynamics = ConfigHydrodynamics()
+    """ Holds the config of the Hydrodynamics class. """
+
+    configThermodynamics: ConfigThermodynamics = ConfigThermodynamics()
+    """ Holds the config of the Thermodynamics class. """
+
+    configBoltzmannSolver: ConfigBoltzmannSolver = ConfigBoltzmannSolver()
+    """ Holds the config of the BoltzmannSolver class. """
+
+    def loadConfigFromFile(self, filePath: str) -> None:
+        """
+        Load the configs from a file.
+
+        Parameters
+        ----------
+        filePath : str
+            Path of the file where the configs are.
+
+        """
+        parser = ConfigParser()
+        parser.readINI(filePath)
+        
+        # Read the Grid configs

```

Comment:
These should all be changed to only modify attributes that were listed in the parsed .ini. So a file that is mostly empty but just has eg.
[Grid] smoothing = ...
would be valid, and loading it would change only smoothing and keep the Config otherwise unchanged.

Can implement by checking if ConfigParser has a specific entry, if not don't update that variable.

I think this is pretty important for the future too, because right now adding a new config variable in Config would require that it's also added to all .inis out there.